server/src/server/competition_info.py

```diff
@@ -0,0 +1,89 @@
+import copy
+import hashlib
+import os
+import typing
+import yaml
+
+from server_types import ChallengeInfoKeys
+
+
+class CompetitionInfo(object):
+    def __init__(self, data_dir: str, event: str):
+        self._event = event
+        event_dir = os.path.join(data_dir, event)
+        self._teams = load_teams(event_dir)
+        self._challenges = load_challenges(event_dir)
+        self._challenge_infos = load_challenge_infos(event_dir, self._challenges)
+
+    @property
+    def event(self) -> str:
+        return self._event
+
+    def list_teams(self) -> typing.List[str]:
+        return list(self._teams)
+
+    def list_challenges(self) -> typing.List[str]:
+        return list(self._challenges)
+
+    def get_challenge_info(self, challenge: str) -> dict:
+        if challenge in self._challenge_infos:
+            return copy.deepcopy(self._challenge_infos[challenge])
+        else:
+            return {
+                ChallengeInfoKeys.NAME: "",
+                ChallengeInfoKeys.DESCRIPTION: "",
+                ChallengeInfoKeys.SCORE_TABLE: [],
+                ChallengeInfoKeys.AVAILABLE_ATTEMPTS: [],
+            }
+
+
+def load_teams(event_dir: str) -> typing.List[str]:
+    with open(os.path.join(event_dir, "teams.yaml"), "r") as f:
+        teams = yaml.safe_load(f)
+    return teams
+
+
+def load_challenges(event_dir: str) -> typing.List[str]:
+    with open(os.path.join(event_dir, "challenges.yaml"), "r") as f:
+        challenges = yaml.safe_load(f)
```

Comment:
Again, I think `safe_load` is outdated?

Reply:
Not that I'm aware, AFAIK this is still preferred. If I use `load`, I get a warning referring me to https://github.com/yaml/pyyaml/wiki/PyYAML-yaml.load(input)-Deprecation .